src/client/action/list_databases.rs

```diff
@@ -0,0 +1,162 @@
+        self
+    }
+
+    /// Filters the query.
+    pub fn filter(mut self, value: impl Into<Option<Document>>) -> Self {
+        self.options().filter = value.into();
+        self
+    }
+
+    /// Determines which databases to return based on the user's access privileges. This option is
+    /// only supported on server versions 4.0.5+.
+    pub fn authorized_databases(mut self, value: impl Into<Option<bool>>) -> Self {
+        self.options().authorized_databases = value.into();
+        self
+    }
+
+    /// Tags the query with an arbitrary [`Bson`] value to help trace the operation through the
+    /// database profiler, currentOp and logs.
+    ///
+    /// This option is only available on server versions 4.4+.
+    pub fn comment(mut self, value: impl Into<Option<Bson>>) -> Self {
+        self.options().comment = value.into();
+        self
+    }
+
+    /// Runs the query using the provided session.
+    pub fn session(mut self, value: impl Into<Option<&'a mut ClientSession>>) -> Self {
+        self.session = value.into();
+        self
+    }
+}
+
+impl<'a> IntoFuture for ListDatabases<'a, Full> {
+    type Output = Result<Vec<DatabaseSpecification>>;
+    type IntoFuture = BoxFuture<'a, Self::Output>;
+
+    fn into_future(self) -> Self::IntoFuture {
+        async {
+            let op = op::ListDatabases::new(false, self.options);
+            self.client.execute_operation(op, self.session).await.and_then(|dbs| {
+                dbs.into_iter()
+                    .map(|db_spec| {
+                        bson::from_slice(db_spec.as_bytes()).map_err(crate::error::Error::from)
+                    })
+                    .collect()
+            })
+        }.boxed()
+    }
+}
+
+impl<'a> IntoFuture for ListDatabases<'a, Names> {
+    type Output = Result<Vec<String>>;
+    type IntoFuture = BoxFuture<'a, Self::Output>;
+
+    fn into_future(self) -> Self::IntoFuture {
+        async {
+            let op = op::ListDatabases::new(true, self.options);
+            match self.client.execute_operation(op, self.session).await {
+                Ok(databases) => databases
+                    .into_iter()
+                    .map(|doc| {
+                        let name = doc
+                            .get_str("name")
+                            .map_err(|_| ErrorKind::InvalidResponse {
+                                message: "Expected \"name\" field in server response, but it was not \
+                                        found"
+                                    .to_string(),
+                            })?;
+                        Ok(name.to_string())
+                    })
+                    .collect(),
+                Err(e) => Err(e),
+            }
+        }.boxed()
+    }
+}
+
+#[cfg(any(feature = "sync", feature = "tokio-sync"))]
+impl<'a, M: Mode> ListDatabases<'a, M>
+    where Self: IntoFuture
+{
+    /// Synchronously execute this action.
+    pub fn run(self) -> <Self as IntoFuture>::Output {
```

Comment:
It's unfortunate that sync users will now always need a `.run()` in there, but I don't see any way around that.